### PR TITLE
Updated list of enabled APIs for network dashboard

### PR DESCRIPTION
**Pull request by maunope**
new list
    "cloudasset.googleapis.com",
    "cloudbilling.googleapis.com",
    "cloudfunctions.googleapis.com",
    "cloudscheduler.googleapis.com",
    "compute.googleapis.com",
    "logging.googleapis.com",
    "monitoring.googleapis.com",
    "pubsub.googleapis.com",
    "servicenetworking.googleapis.com",
    "storage-component.googleapis.com"

### CODE REVIEW

**Review by aurelienlegrand — APPROVED** [review]
LGTM

File: blueprints/cloud-operations/network-dashboard/variables.tf [hunk 1]
```diff
@@ -50,28 +50,35 @@ variable "organization_id" {
 
 variable "prefix" {
   description = "Customer name to use as prefix for monitoring project"
+  default     = "mon"
```
Comment (juliocc) [inline blueprints/cloud-operations/network-dashboard/variables.tf:53]: We usually don't provide defaults for prefix

File: blueprints/cloud-operations/network-dashboard/variables.tf [hunk 2]
```diff
@@ -50,28 +50,35 @@ variable "organization_id" {
...
     "run.googleapis.com",
-    "serviceusage.googleapis.com"
+    "servicenetworking.googleapis.com",
+    "serviceusage.googleapis.com",
+    "storage-component.googleapis.com"
   ]
 }
 
+
+
```
Comment (juliocc) [inline blueprints/cloud-operations/network-dashboard/variables.tf:81]: There's a few new empty lines in the file. Can you keep a single empty line between variables?

File: blueprints/cloud-operations/network-dashboard/README.md [hunk 3]
```diff
@@ -16,13 +16,14 @@ Three metric descriptors are created for each monitored resource: usage, limit a
...
-  v2                      = false                     # Set to true to use V2 Cloud Functions environment
-  ```
+  - organization_id = "<YOUR-ORG-ID>"
+  - billing_account = "<YOUR-BILLING-ACCOUNT>"
+  - monitoring_project_id = "<YOUR-MONITORING-PROJECT>" # Monitoring project where the dashboard will be created and the solution deployed, a project named "mon-network-dahshboard" will be created if left blank
+  - monitored_projects_list = ["project-1", "project2"] # Projects to be monitored by the solution
+  - monitored_folders_list = ["folder_id"] # Folders to be monitored by the solution
+  - prefix = "<YOUR-PREFIX>" # Monitoring project name prefix, monitoring project name is <YOUR-PREFIX>-network-dashboard, ignored if monitoring_project_id variable is provided
+  - v2 = true|false # Set to true to use V2 Cloud Functions environment
+    ```
```
Comment (juliocc) [inline blueprints/cloud-operations/network-dashboard/README.md:26]: The inline comments are kind of long. Maybe you could put them in their own line?

File: blueprints/cloud-operations/network-dashboard/README.md [hunk 4]
```diff
@@ -16,13 +16,14 @@ Three metric descriptors are created for each monitored resource: usage, limit a
...
 - Create a terraform.tfvars file with the following content:
   ```tfvars
-  organization_id         = "<YOUR-ORG-ID>"
-  billing_account         = "<YOUR-BILLING-ACCOUNT>"
-  monitoring_project_id   = "project-0"               # Monitoring project where the dashboard will be created and the solution deployed
-  monitored_projects_list = ["project-1", "project2"] # Projects to be monitored by the solution
-  monitored_folders_list  = ["folder_id"]             # Folders to be monitored by the solution
-  v2                      = false                     # Set to true to use V2 Cloud Functions environment
-  ```
+  - organization_id = "<YOUR-ORG-ID>"
```
Comment (juliocc) [inline blueprints/cloud-operations/network-dashboard/README.md:19]: I would remove the leading dash. As it it stands, it's neither a tfvars file nor easy to copy/paste.

**Review by juliocc — CHANGES_REQUESTED** [review]

File: blueprints/cloud-operations/network-dashboard/variables.tf (hunk 1 above)
Comment (maunope) [reply blueprints/cloud-operations/network-dashboard/variables.tf:60]: kk

File: blueprints/cloud-operations/network-dashboard/variables.tf (hunk 2 above)
Comment (maunope) [reply blueprints/cloud-operations/network-dashboard/variables.tf:88]: kk

File: blueprints/cloud-operations/network-dashboard/README.md (hunk 4 above)
Comment (maunope) [reply blueprints/cloud-operations/network-dashboard/README.md:19]: kk

File: blueprints/cloud-operations/network-dashboard/README.md [hunk 5]
```diff
@@ -16,23 +16,28 @@ Three metric descriptors are created for each monitored resource: usage, limit a
...
+  billing_account = "<YOUR-BILLING-ACCOUNT>"
+  monitoring_project_id = "<YOUR-MONITORING-PROJECT>" 
+  # Monitoring project where the dashboard will be created and the solution deployed, a project named "mon-network-dahshboard" will be created if left blank
+  monitored_projects_list = ["project-1", "project2"] 
+  # Projects to be monitored by the solution
+  monitored_folders_list = ["folder_id"] 
+  # Folders to be monitored by the solution
+  prefix = "<YOUR-PREFIX>" 
+  # Monitoring project name prefix, monitoring project name is <YOUR-PREFIX>-network-dashboard, ignored if monitoring_project_id variable is provided
+  v2 = true|false # Set to true to use V2 Cloud Functions environment
```
[the remark this answers is not in the document]
Comment (maunope) [reply blueprints/cloud-operations/network-dashboard/README.md:29]: kk

File: blueprints/cloud-operations/network-dashboard/README.md [hunk 6]
```diff
@@ -16,23 +16,29 @@ Three metric descriptors are created for each monitored resource: usage, limit a
...
+  cf_version = V1|V2
+  # Set to V2 to use V2 Cloud Functions environment
+    ```
 - `terraform init`
 - `terraform apply`
 
 Note: Org level viewing permission is required for some metrics such as firewall policies.
 
-Once the resources are deployed, go to the following page to see the dashboard: https://console.cloud.google.com/monitoring/dashboards?project=<YOUR-MONITORING-PROJECT>.
+Once the resources are deployed, go to the following page to see the dashboard: https://console.cloud.google.com/monitoring/dashboards?project=<YOUR-MONITORING-PROJECT> (or <YOUR-METRICS-PROJECT> if populated)
```
Comment (aurelienlegrand) [inline blueprints/cloud-operations/network-dashboard/README.md:37]: Is this change needed?

File: blueprints/cloud-operations/network-dashboard/README.md [hunk 7]
```diff
@@ -16,23 +16,29 @@ Three metric descriptors are created for each monitored resource: usage, limit a
...
 A dashboard called "quotas-utilization" should be created.
 
 The Cloud Function runs every 10 minutes by default so you should start getting some data points after a few minutes.
-You can use the metric explorer to view the data points for the different custom metrics created: https://console.cloud.google.com/monitoring/metrics-explorer?project=<YOUR-MONITORING-PROJECT>.
+You can use the metric explorer to view the data points for the different custom metrics created: https://console.cloud.google.com/monitoring/metrics-explorer?project=<YOUR-MONITORING-PROJECT> (or <YOUR-METRICS-PROJECT> if populated).
```
Comment (aurelienlegrand) [inline blueprints/cloud-operations/network-dashboard/README.md:41]: Same

File: blueprints/cloud-operations/network-dashboard/main.tf [hunk 8]
```diff
@@ -21,6 +21,7 @@ locals {
   folder_ids         = toset(var.monitored_folders_list)
   folders            = join(",", local.folder_ids)
   monitoring_project = var.monitoring_project_id == "" ? module.project-monitoring[0].project_id : var.monitoring_project_id
+  metrics_project    = var.metrics_project_id == "" ? (var.monitoring_project_id == "" ? module.project-monitoring[0].project_id : var.monitoring_project_id) : var.metrics_project_id
```
Comment (aurelienlegrand) [inline blueprints/cloud-operations/network-dashboard/main.tf:24]: Is this needed or part of another PR?

File: blueprints/cloud-operations/network-dashboard/main.tf [hunk 9]
```diff
@@ -182,5 +190,5 @@ module "cloud-function" {
 
 resource "google_monitoring_dashboard" "dashboard" {
   dashboard_json = file("${path.module}/dashboards/quotas-utilization.json")
-  project        = local.monitoring_project
+  project        = local.metrics_project
```
Comment (aurelienlegrand) [inline blueprints/cloud-operations/network-dashboard/main.tf:193]: same

File: blueprints/cloud-operations/network-dashboard/README.md (hunk 6 above)
Comment (maunope) [reply blueprints/cloud-operations/network-dashboard/README.md:37]: moving to next pr

File: blueprints/cloud-operations/network-dashboard/main.tf (hunk 9 above)
Comment (maunope) [reply blueprints/cloud-operations/network-dashboard/main.tf:193]: moving to next pr

File: blueprints/cloud-operations/network-dashboard/main.tf (hunk 8 above)
Comment (maunope) [reply blueprints/cloud-operations/network-dashboard/main.tf]: removed

**Review by juliocc — APPROVED** [review]
